### DISCUSSION TITLE
ValidationError fail OpenAPI Schema validation

### DISCUSSION THREAD

**Discussion by okamop** [fastapi/fastapi#6387] [Questions] [answered]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.get("/")
async def root():
    return {"message": "Hello World"}
```

### Description

* Running [`openapi-generator`](https://github.com/OpenAPITools/openapi-generator) against the schema generated by FastAPI fails.
* `ValidationError` schema cannot be processed.
* Generation fails with FastAPI v0.75.0 or higher
* Generated successfully with FastAPI v0.74.1

```
$ openapi-generator-cli generate -g typescript-axios -i http://localhost:8000/openapi.json

[main] INFO  o.o.codegen.DefaultGenerator - Generating with dryRun=false
[main] INFO  o.o.codegen.DefaultGenerator - OpenAPI Generator: typescript-axios (client)
[main] INFO  o.o.codegen.DefaultGenerator - Generator 'typescript-axios' is considered stable.
[main] INFO  o.o.c.l.AbstractTypeScriptClientCodegen - Hint: Environment variable 'TS_POST_PROCESS_FILE' (optional) not defined. E.g. to format the source code, please try 'export TS_POST_PROCESS_FILE="/usr/local/bin/prettier --write"' (Linux/Mac)
[main] INFO  o.o.c.l.AbstractTypeScriptClientCodegen - Note: To enable file post-processing, 'enablePostProcessFile' must be set to `true` (--enable-post-process-file for CLI).
[main] INFO  o.o.codegen.utils.URLPathUtils - 'host' (OAS 2.0) or 'servers' (OAS 3.0) not defined in the spec. Default to [http://localhost] for server URL [http://localhost/]
[main] INFO  o.o.codegen.utils.URLPathUtils - 'host' (OAS 2.0) or 'servers' (OAS 3.0) not defined in the spec. Default to [http://localhost] for server URL [http://localhost/]
[main] WARN  o.o.c.l.AbstractTypeScriptClientCodegen - The import is a union type. Consider using the toModelImportMap method.
Exception in thread "main" java.lang.RuntimeException: Could not process model 'ValidationError'.Please make sure that your schema is correct!
        at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:492)
        at org.openapitools.codegen.DefaultGenerator.generate(DefaultGenerator.java:888)
        at org.openapitools.codegen.cmd.Generate.execute(Generate.java:441)
        at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
        at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
Caused by: java.lang.StringIndexOutOfBoundsException: begin 1, end 0, length 0
        at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3751)
        at java.base/java.lang.String.substring(String.java:1907)
        at java.base/java.lang.String.substring(String.java:1880)
        at org.openapitools.codegen.languages.TypeScriptAxiosClientCodegen.postProcessModels(TypeScriptAxiosClientCodegen.java:233)
        at org.openapitools.codegen.DefaultGenerator.processModels(DefaultGenerator.java:1302)
        at org.openapitools.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:487)
        ... 4 more

Process finished with exit code 1
```

### Operating System

macOS

### FastAPI Version

0.75.2

### Python Version

3.10.4

**Comment by andvikt** [comment]
same on 0.76.0

**Comment by vhuliai-beetroot** [comment]
Same on `0.77.1`

**Comment by flxdot** [comment, marked as answer]
Well, it is due to this change: https://github.com/tiangolo/fastapi/pull/4806

But I would rather argue this is an issue of [openapi-generator](https://github.com/OpenAPITools/openapi-generator), as the schema is valid according to the specification of OpenAPI. There is even an open issue for that: https://github.com/OpenAPITools/openapi-generator/issues/4576

**Comment by okamop** [comment]
> Well, it is due to this change: #4806
> 
> But I would rather argue this is an issue of [openapi-generator](https://github.com/OpenAPITools/openapi-generator), as the schema is valid according to the specification of OpenAPI. There is even an open issue for that: [OpenAPITools/openapi-generator#4576](https://github.com/OpenAPITools/openapi-generator/issues/4576)

It seems so.
I used [openapi-typescript](https://github.com/drwpow/openapi-typescript#readme) instead.
This one generates without any problem.

Close this issue.

**Comment by bykof** [comment]
I used the [typescript-fetch](https://openapi-generator.tech/docs/generators/typescript-fetch) generator and it worked.

**Comment by gaganpreet** [comment]
This issue with typescript-axios generator is fixed in the main branch of openapi generator, but it's not packaged yet. Using git bisect I pinned the fix down to [this commit](https://github.com/OpenAPITools/openapi-generator/commit/8330e16d663b722e47a5fcf47c8af6e679ab752a)

Edit: Downgrading to 5.3.x also works.

**Comment by FFengIll** [comment]
I have to show that new version of fastapi output openapi on version 3.1.0,
but swagger-codegen only work on 3.0.x.

I use bellow to solve it
```python
if __name__ == "__main__":
    with open("./data/swagger.json", "w") as fd:
        # though the fastapi support 3.1.0, but swagger-codegen not
        # force downgrade it since compatibility
        app.openapi_version = "3.0.2"
        schema = app.openapi()
        # schema["openapi"] = "3.0.2"
        json.dump(schema, fd, indent=4, ensure_ascii=False)
```
